src/utils/py/README.md

```diff
@@ -1 +1,244 @@
 # Python Bindings for DAOS API
+
+## Motivation for Python API
+
+The Python API for DAOS provides access to DAOS API functionality with an emphasis on test use cases. While the majority of unit tests are written in C, higher-level tests are written primarily using the Python API. Interfaces are provided for accessing DAOS management and DAOS API functionality from Python. This higher level interface allows a faster turnaround time on implementing test cases for DAOS.
+
+## Architecture
+
+### Layout
+
+The Python API is split into several files based on functionality:
+
+  * The Python object API, located in the source tree at [src/utils/py/daos_api.py](daos_api.py).
+  * The mapping of C structures to Python classes, located in the source tree at [src/utils/py/daos_cref.py](daos_cref.py)
+
+High-level abstraction classes exist to manipulate DAOS storage:
+```python
+class DaosPool(object)
+class DaosContainer(object)
+class DaosObj(object)
+class IORequest(object)
+```
+
+`DaosPool` is a Python class representing a DAOS pool. All pool-related functionality is exposed from this class. Operations such as creating/destroying a pool, connecting to a pool, and adding a target to a storage pool are supported.
+
+`DaosContainer` is a Python class representing a DAOS container. As with the `DaosPool` class, all container-related functionality is exposed here. This class also exposes abstracted wrapper functions for the flow of creating and committing an object to a DAOS container.
+
+`DaosObj` is a Python class representing a DAOS object. Functionality such as creating/deleting objects in a container, 'punching' objects (delete an object from the specified transaction only), and object query.
+
+`IORequest` is a Python class representing a DAOS transaction. Functionality is exposed to create, delete, and modify DAOS transactions.
```

Comment:
IORequest is a read or write request against a DAOS object not a transaction.  The IORequest is processed within the context of a transaction but you can process many such requests in a single transaction.  I may have said this better in previous comments.

Reply:
Fixed phrasing to match this more closely.